File: Question/27_Remove_Element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        if(nums.size() == 0) return 0;
        int left = 0;
        int right = 0;
        while(right < nums.size()) {
            if(nums[right] == val) {
                ++right;
            } else {
                nums[left++] = nums[right++];
            }
        }
        return left;
    }
};

File: Question_Java/27_Remove_Element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int cur = 0;
        for (int i = 0; i < nums.length; ++i) {
            if (nums[i] == val)
                continue;
            nums[cur++] = nums[i];
        }
        return cur;
    }
}